temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -139,11 +147,15 @@ public boolean start() {
   @Override
   public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
     return poller.shutdown(shutdownManager, interruptTasks);
+    // TODO wait for returning all slots to semaphore to make sure all eager dispatch handlers are
+    // respected
   }
 
   @Override
   public void awaitTermination(long timeout, TimeUnit unit) {
     ShutdownManager.awaitTermination(poller, unit.toMillis(timeout));
+    // TODO wait for returning all slots to semaphore to make sure all eager dispatch handlers are
+    // respected
   }
```

Comment:
Are these going to be addressed in the next PR or are you going to open issues?

Also, pardon my ignorance, but why do we need to wait both in awaitTermination and shutdown?

Reply:
> Are these going to be addressed in the next PR or are you going to open issues?

I will do it in a separate PR. It's a larger scope than it seems because it will need to change/fix the shutdown implementation of a Poller. 

> Also, pardon my ignorance, but why do we need to wait both in awaitTermination and shutdown?

`shutdown` will wait for semaphore to perform the last stage of shutdown after the mentioned PR and change.
`awaitTermination` will wait for that last state to be executed.